src/main/java/org/opensearch/searchrelevance/executors/LlmJudgmentTaskManager.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.searchrelevance.executors;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Semaphore;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import org.opensearch.common.inject.Inject;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.searchrelevance.exception.SearchRelevanceException;
+import org.opensearch.searchrelevance.judgments.JudgmentDataTransformer;
+import org.opensearch.threadpool.ThreadPool;
+
+import lombok.extern.log4j.Log4j2;
+
+/**
+ * Manages concurrent execution of LLM judgment tasks at the query text level
+ */
+@Log4j2
+public class LlmJudgmentTaskManager {
+    private static final String THREAD_POOL_EXECUTOR_NAME = ThreadPool.Names.GENERIC;
```

Comment:
cool that we're using generic pool, not new dedicated one. With current approach system should be more stable under heavy load